wordpress-mobile/WordPress-Android
# [Prompts] Use Blogging Prompts v3 Endpoint

Fixes #18257 
FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2721

Point to FluxC version using Prompts v3. The data models changes, requiring several changes. A new field is the `answeredLink` which we are now using to get the prompt ID tag for both the Reader ("View all responses") and the tags inside the editor when answering a prompt.

There's an [open backend PR](https://github.com/Automattic/jetpack/pull/30255) at the moment that is required to force the dates to what we expect the endpoint to respond, so testing this PR is a bit tricky at the moment, requiring pulling in the backend jetpack plugin changes in a sandbox environment and point/proxy the testing device to use that sandbox for the `public-api.wordpress.com` calls. This is also the reason this PR is `NOT READY FOR MERGE` yet.

Without doing the above, a clean install of this PR will not show the prompts card.

To test:
Once the backend changes are deployed to production **or** the reviewer sets up the sandbox and pulls the backend PR mentioned above, these are the steps:

1. Open Jetpack app
2. Login with a WP.com account
3. Select a blogging site (has a few posts)
4. Go to `My Site > Home`
5. **Verify** the prompts card show up
6. **Verify** the prompt text does **not** match the v2 prompt, which is the one shown on older versions or the web
7. Tap "Answer prompt"
8. **Verify** the prompt quote block matches the prompt text expected
9. Tap the overflow menu and go to "Post settings"
10. **Verify** the post has 2 tags: "dailyprompt" and "dailyprompt-<id>"
11. Go back to `My Site > Home`
12. On the prompts card, tap "View all responses"
13. **Verify** the topic tag matches the expected "dailyprompt-<id>" tag
14. Go back
15. Tap the overflow menu in the prompts card, then tap "View more prompts"
16. **Verify** everything works as expected

## Regression Notes
1. Potential unintended areas of impact
Dashboard breakage.

17. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

18. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests.

PR submission checklist:


UI Changes testing checklist:

**N/A, this didn't introduce any UI changes**
